Sources/NIOCore/EventLoop.swift

```diff
@@ -356,6 +356,14 @@ public protocol EventLoop: EventLoopGroup {
     /// It is valid for an `EventLoop` not to implement any of the two `_promise` functions. If either of them are implemented,
     /// however, both of them should be implemented.
     func _promiseCompleted(futureIdentifier: _NIOEventLoopFutureIdentifier) -> (file: StaticString, line: UInt)?
+
+    /// Set a timer that will call a handler at the given time.
+    @discardableResult
+    func setTimer(for deadline: NIODeadline, _ handler: any NIOTimerHandler) -> NIOTimer
+
+    /// Set a timer that will call a handler after a given amount of time.
+    @discardableResult
+    func setTimer(for duration: TimeAmount, _ handler: any NIOTimerHandler) -> NIOTimer
```

Comment:
While I'm here, can I suggest that we should use different labels instead of `for` in both? That will help with tab-completion a bit.

Reply:
Hm, it felt OK to me. Would you prefer the following?

- `setTimer(for: TimeAmount, ...)`
- `setTimer(forDeadline: NIODeadline, ...)`

I'm thinking of how I set timers verbally on my phone I normally always use "set a timer _for_" regardless of whether I then say a duration or an absolute time.

Open to suggestions here, though :)

Reply:
`setTimer(for:)` and `setTimer(at:)` is probably the easiest spelling distinction.

Reply:
https://github.com/apple/swift-nio/pull/2759/files/df3cdb1c1f1f3ce8fb66fd81cdfa0cef3318c984#r1664313726 resulted in new API spellings. But these do use different prepositions for the `NIODeadline` and `TimeAmount` variants: `scheduleCallback(at:handler:)` and `scheduleCallback(in:handler:)`, respectively.